### PR TITLE
chore: sync with osx-plugin-template-hardhat develop branch 

### PR DESCRIPTION
**Pull request by heueristik**
This PR brings in the changes from https://github.com/aragon/osx-plugin-template-hardhat/tree/develop

Task ID: [OS-1145](https://aragonassociation.atlassian.net/browse/OS-1145)

[OS-1145]: https://aragonassociation.atlassian.net/browse/OS-1145?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/subgraph/manifest/data/arbitrum.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  "dataSources": {
+    "PluginSetupProcessors": [
+      {
+        "name": "PluginSetupProcessor",
+        "address": "0x308a1DC5020c4B5d992F5543a7236c465997fecB",
+        "startBlock": 145462184
+      }
+    ],
+    "Plugin": {
+      "name": "MyPlugin",
```
Comment (heueristik) [inline packages/subgraph/manifest/data/arbitrum.json:13]: Do these need to be renamed @jordaniza ?

Comment (jordaniza) [reply packages/subgraph/manifest/data/arbitrum.json:13]: I'll do it when we deploy

**Review by jordaniza — APPROVED** [review]

File: packages/contracts/utils/helpers.ts [hunk 2]
```diff
@@ -188,6 +195,17 @@ export async function createVersion(
...
+  const allowedCharacters = 'abcdefghijklmnopqrstuvwxyz-0123456789';
+  let result = '';
+  for (let i = 0; i < length; i++) {
+    result += allowedCharacters.charAt(
+      Math.floor(Math.random() * allowedCharacters.length)
+    );
+  }
+  return result;
+}
+
```
Comment (clauBv23) [inline packages/contracts/utils/helpers.ts:208]: this function shouldn’t be needed right? The plugin name [is not randomly generated](https://github.com/aragon/token-voting-plugin/blob/7172dee5b4f143ce6ce218a552764895909b8e0e/packages/contracts/plugin-settings.ts#L9)

File: packages/subgraph/manifest/data/arbitrum.json (hunk 1 above)
Comment (clauBv23) [reply packages/subgraph/manifest/data/arbitrum.json:13]: I think we could remove the name and the `startBlock`, only the address is needed. 

The `startBlock` also is misleading, it is just a random number, it should be at least changed

**Review by clauBv23 — APPROVED** [review]

**Pull request merged**